src/CLMWrapper.cpp

```diff
@@ -576,11 +570,30 @@ void ClmWrapper::callback(const sensor_msgs::ImageConstPtr& msgIn)
     active_models_st += active_m_C;
     cv::putText(disp_image, active_models_st, cv::Point(10,60), CV_FONT_HERSHEY_SIMPLEX, 0.5, CV_RGB(255,0,0));   
 
-    if(!clm_parameters[0].quiet_mode)
+    if (faceDetected)
     {
-        namedWindow("tracking_result",1);   
-        cv::imshow("tracking_result", disp_image);
+        // feeding disp_image into the cv pointer's image
+        // only if a face is detected
+        cv::cvtColor(disp_image,cv_ptr->image, CV_RGB2BGR);
     }
+     //publishing the image usign the cv pointer
+    imagePublisher.publish(cv_ptr->toImageMsg());
```

Comment:
Yes exactly. What I usually do is to declare a function like this that does the job for me (is taken from another repo but you can get the idea):

```
bool BoardCalibrator::publishImage(cv::Mat &mat, const std::string encoding)
{
    cv_bridge::CvImage msgOut;
    msgOut.encoding = encoding;
    msgOut.image    = mat;

    imagePublisher.publish(msgOut.toImageMsg());
    return true;
};
```

The initial cv_ptr should only take care of the initial conversion.